app-android/src/main/java/com/ustadmobile/port/android/view/MessagesRecyclerAdapter.kt

```diff
@@ -0,0 +1,97 @@
+import com.ustadmobile.lib.db.entities.MessageWithPerson
+import com.ustadmobile.port.android.view.util.SelectablePagedListAdapter
+import me.saket.bettermovementmethod.BetterLinkMovementMethod
+
+
+class MessagesRecyclerAdapter(val loggedInPersonUid: Long)
+    : SelectablePagedListAdapter<MessageWithPerson,
+        MessagesRecyclerAdapter.MessageWithPersonViewHolder>(DIFF_CALLBACK_COMMENTS) {
+
+    var presenter: ChatDetailPresenter? = null
+
+    class MessageWithPersonViewHolder(val binding: ItemMessageListBinding)
+        : RecyclerView.ViewHolder(binding.root)
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MessageWithPersonViewHolder {
+        val vh = MessageWithPersonViewHolder(ItemMessageListBinding.inflate(
+                LayoutInflater.from(parent.context), parent, false))
+
+        addMovement(vh.binding.itemCommentsListLine2Text, parent.context)
+
+        return vh
+    }
+
+
+    override fun onBindViewHolder(holder: MessageWithPersonViewHolder, position: Int) {
+        if(itemCount > 0 ) {
+
+            val message = getItem(position)
+            holder.binding.loggedInPersonUid = loggedInPersonUid
+            holder.itemView.tag = message?.messageUid
+            holder.binding.message = message
+
+            holder.binding.itemCommentsListLine2Text.text = message?.messageText
+            addMovement(holder.binding.itemCommentsListLine2Text, holder.itemView.context)
+
+        }
+    }
+
+
+    private fun addMovement(textView: TextView, context: Context){
+
+        textView.linksClickable = true
+
+        textView.movementMethod =
+            BetterLinkMovementMethod.newInstance().apply {
+                this.setOnLinkClickListener(onClickListener)
+                this.setOnLinkLongClickListener(onLongClickListener)
+
+            }
+
+        Linkify.addLinks(textView, Linkify.ALL)
+
+    }
+
+    val onClickListener: BetterLinkMovementMethod.OnLinkClickListener =
+        BetterLinkMovementMethod.OnLinkClickListener{
+        _,url ->
+        presenter?.handleClickLink(url)
+
+        true
+    }
+
+    val onLongClickListener: BetterLinkMovementMethod.OnLinkLongClickListener =
+        BetterLinkMovementMethod.OnLinkLongClickListener{
+            _,_ ->
+        true
+    }
+
+    companion object{
+
+        val DIFF_CALLBACK_COMMENTS =
+                object : DiffUtil.ItemCallback<MessageWithPerson>() {
+                    override fun areItemsTheSame(oldItem: MessageWithPerson,
+                                                 newItem: MessageWithPerson): Boolean {
+                        return oldItem.messageUid == newItem.messageUid
+                    }
+
+                    override fun areContentsTheSame(oldItem: MessageWithPerson,
+                                                    newItem: MessageWithPerson): Boolean {
+                        return oldItem.messageSenderPersonUid == newItem.messageSenderPersonUid &&
```

Comment:
I think we should check the person name here as well. That is what is displayed. The name wouldn't change very often, but it is part of what is shown to the user.